engine/src/main/java/org/destinationsol/game/SolObject.java

```diff
@@ -16,41 +16,187 @@
+     * crosses for ships and stuff alike.
+     *
+     * @param game Game this object belongs to.
+     */
+    void update(SolGame game);
 
+    /**
+     * Denotes whether the object should be removed as of the time of calling.
+     * <p>
+     * Object should usually be removed when its health reaches zero, or, for instance, if the object is projectile, it
+     * reaches its target. This method should not handle removal of object for optimization purposes when it gets too
+     * far away, unless the object is specifically designed to exist only in close proximity of player, like some sort
+     * of kinetic defense. Removal for optimization purposes is handled in {@link #toFarObject()}.
+     *
+     * @param game Game this object belongs to.
+     * @return Boolean denoting whether the object should be removed.
+     */
+     boolean shouldBeRemoved(SolGame game);
+
+    /**
+     * Called whenever an object is due to be removed from game.
+     * <p>
+     * This method should handle things like throwing out {@link Loot} when ships are destroyed, or creating smaller
+     * asteroids when a big {@link Asteroid} is destroyed. This method is also responsible for freeing its resources,
+     * when such is applicable. Note that this method is also called when an object is due to be removed for
+     * optimization purposes, and thus you need to explicitly check for {@link #shouldBeRemoved(SolGame)} if you have
+     * some code specific to destruction of object.
+     *
+     * @param game Game this object belongs to.
+     */
     void onRemove(SolGame game);
 
+    /**
+     * Called whenever this object should receive damage.
+     * <p>
+     * Whenever the object is hit by some projectile or other sort of damage dealing thing, this method is called on the
+     * object to allow it to perform its damage handling. This usually means just subtracting the damage from object's
+     * health and playing a hit sound, but if the object has no health pool or should be otherwise indestructible, or
+     * invincible against some types of damage, this method can be freely left blank.
+     *
+     * @param dmg      Damage the object receives.
+     * @param game     Game this object belongs to.
+     * @param position Position the object was hit at, if hit by point-based damage. Null if not applicable, such as fire.
+     * @param dmgType  Type of the damage object receives.
+     */
     void receiveDmg(float dmg, SolGame game, @Nullable Vector2 position, DmgType dmgType);
 
+    /**
+     * Denotes whether this object is affected by gravity, this is usually constant.
+     *
+     * @return True if object is affected by gravity, false otherwise.
+     */
     boolean receivesGravity();
 
+    /**
+     * Applies external force to object.
+     * <p>
+     * This method usually just scales the {@code force} by objects mass if {@code acc} is set, and then passes the
+     * force to object's internal {@link Body}.
```

Comment:
`* If {@code acc} is true, the supplied {@code force} is treated as an acceleration and considered to not have been scaled by mass.` We do not need `, and then passes the force to object's internal {@link Body}.`

Side note: I think that the `acc` parameter is redundant and just leads to more confusion. The method, after all, is called `receiveForce` and not `receiveAcceleration`! This may be something to address in a later PR.